### PR TITLE
[onprem] HDFS Interface

### PR DESCRIPTION
**Pull request by HaileyJang**
Continued from previous PR

### CODE REVIEW

File: skyplane/obj_store/hdfs_interface.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        try:
+            self.get_obj_metadata(obj_name)
+            return True
+        except NoSuchObjectException:
+            return False
+        
+    def region_tag(self) -> str:
+        return ""
+
+    def bucket(self) -> str:
```
Comment (HaileyJang) [inline skyplane/obj_store/hdfs_interface.py:41]: In the previous meeting, we said that we can think of buckets as directories. However, considering that we are stateless with files, would keeping track of the entered directory (when they initially initiate the connection) be okay?

File: tests/unit_nocloud/test_hdfs.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+from skyplane.obj_store.hdfs_interface import HDFSFile, HDFSInterface
```
Comment (HaileyJang) [inline tests/unit_nocloud/test_hdfs.py:1]: Unable to find hdfs_interface module when I try to run this test. Probably have to add the HDFS Interface to the Object Store create function.

File: skyplane/obj_store/hdfs_interface.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        return ""
+        
+    def create_bucket(self, region_tag: str):
+        return None
+
+    def delete_bucket(self):
+        return None
+
+    def bucket_exists(self) -> bool:
```
Comment (HaileyJang) [inline skyplane/obj_store/hdfs_interface.py:50]: Where would we need this function in the scope of HDFS? If the bucket (the directory) didn't exist and considered the "bucket" as the folder that the connection is entered into, it wouldn't be able to make the connection from the start if it didn't exist.

**Comment by gilv** [comment]
@HaileyJang how you will obtain parallelism when reading data from HDFS?

**Comment by ShishirPatil** [comment]
> @HaileyJang how you will obtain parallelism when reading data from HDFS?

Hey @gilv for now, the parallelism is restricted to reading multiple files concurrently. Obviously, this isn't very smart as we are bottlenecked by the name node. Once we have an end 2 end system and a good understanding of how this is impacting the performance we plan to revisit it. If you have thoughts, happy to hear!

**Comment by gilv** [comment]
@HaileyJang HDFS can perfectly deal with many reads and name node is never an issue, as all cached in memory. I would suggest you to extend hadoop-distcp. This way you will get perfect parallelism over HDFS cluster and then use your techniques to efficiently copy data

**Pull request merged**